pandas/core/indexes/datetimes.py

```diff
@@ -540,7 +541,10 @@ def _validate_partial_date_slice(self, reso: str):
 
     def _maybe_promote(self, other):
         if other.inferred_type == "date":
-            other = DatetimeIndex(other)
+            try:
+                other = DatetimeIndex(other)
+            except OutOfBoundsDatetime:
```

Comment:
what is hitting this? why are we not bubbling this up?

Reply:
Cases where we have a datetime.date that's out of bounds for DatetimeIndex (e.g. https://github.com/pandas-dev/pandas/pull/33749/files#diff-0c5fadf83f91d24a931a051d82801917R569) are raising here instead of returning -1  which I believe is a bug on master (that's what @jschendel and I were talking about above). I don't think we'd want this to actually raise since it doesn't seem to be an essential step.